src/realm/alloc_slab.cpp

```diff
@@ -300,7 +300,6 @@ SlabAlloc::FreeBlock* SlabAlloc::pop_freelist_entry(FreeList list)
 
 void SlabAlloc::FreeBlock::unlink()
 {
-    REALM_ASSERT_DEBUG(next != nullptr && prev != nullptr);
```

Comment:
I don't think we should delete this. It is good for catching some memory corruption.